src/SmartComponents/ActivityTable/ActivityTable.js

```diff
@@ -62,10 +66,26 @@ const ActivityTable = () => {
   };
 
   const fetchData = async () => {
-    const path = `?limit=500&offset=0`;
-    const result = await fetchExecutedTasks(path);
+    let results;
+    const batchSize = 200;
+    const pages = Math.ceil(totalTaskCount / batchSize) || 1;
+    const result = await resolve(
+      [...new Array(pages)].map(
+        (_, pageIdx) => () =>
+          fetchExecutedTasks(
+            `?limit=${batchSize}&offset=${batchSize * pageIdx}`
+          )
+      )
+    );
+
+    if (isError(result[0])) {
```

Comment:
Nitpick: what happens if the API returns an error? I think the UI breaks because of this array's zero index access. It would be great if we handle this case.

Reply:
This handles that error. Unfortunately, if the api returns an error, it would continue to fetch and return an error for each item instead of stopping the fetch. So this if statement grabs the first index which would include the error details. I tested this by providing a bad url to the api and the if statement handled the issue. But if the first api call succeeds and another doesn't, it might not work right. The alternative would be to loop through each result and check for errors.

Reply:
I'll create a card for this to fix later. I want to get this into prod asap.